### PR TITLE
Update JavaDoc comments

### PR DESCRIPTION
**Pull request by sangonzal**
#169 and #75

### CODE REVIEW

File: src/main/java/com/microsoft/aad/msal4j/AuthorizationCodeParameters.java [hunk 1]
```diff
@@ -22,16 +22,27 @@
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class AuthorizationCodeParameters {
 
+    /**
+     * Authorization code acquired in the first step of OAuth2.0 authorization code flow. For more
+     * details, see https://aka.ms/msal4j-authorization-code-flow
```
Comment (navyasric) [inline src/main/java/com/microsoft/aad/msal4j/AuthorizationCodeParameters.java]: Can we update this aka.ms link to 
https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-acquire-token?tabs=java#acquire-a-token-interactively

File: src/main/java/com/microsoft/aad/msal4j/AuthenticationErrorCode.java [hunk 2]
```diff
@@ -9,19 +9,21 @@
 public class AuthenticationErrorCode {
 
     /**
-     * In the context of device code user has not yet authenticated via browser
+     * In the context of device code user has not yet authenticated via browser. For more details,
+     * see https://aka.ms/msal4j-device-code
```
Comment (navyasric) [inline src/main/java/com/microsoft/aad/msal4j/AuthenticationErrorCode.java]: Can we update this aka.ms link to https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-acquire-token?tabs=java#command-line-tool-without-a-web-browser

File: src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java [hunk 3]
```diff
@@ -10,7 +10,8 @@
 import java.security.cert.X509Certificate;
 
 /**
- * Factory for creating client credentials used in confidential client flows
+ * Factory for creating client credentials used in confidential client flows. For more details, see
+ * https://aka.ms/msal4j-client-credentials
```
Comment (navyasric) [inline src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java]: Can we update this aka.ms link to https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-daemon-acquire-token?tabs=java#acquiretokenforclient-api

File: src/main/java/com/microsoft/aad/msal4j/UserNamePasswordParameters.java [hunk 4]
```diff
@@ -14,19 +14,30 @@
 /**
  * Object containing parameters for Username/Password flow. Can be used as parameter to
  * {@link PublicClientApplication#acquireToken(UserNamePasswordParameters)}
+ *
+ * For more details, see https://aka.ms/msal4j-username-password
```
Comment (navyasric) [inline src/main/java/com/microsoft/aad/msal4j/UserNamePasswordParameters.java]: Can we update aka.ms link to https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-acquire-token?tabs=java#username-and-password

File: src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java (hunk 3 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java]: The docs link you're pointing to doesn't say anything about ClientCredentialFactory (how to initialize a client credential for confidential clients). The wiki page is more useful.

**Comment by sangonzal** [comment]
@navyasric Thanks for reviewing. Yes I plan on changing the redirect links to point to the microsoft docs (other than the client-credentials link) when we update the code snippets.

File: src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java (hunk 3 above)
Comment (navyasric) [reply src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java]: Based on the new patterns in the dev samples, we are going to update docs pages. We can probably update the links once the docs pages are updated.

Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/ClientCredentialFactory.java]: Sounds good, I'm fine with that.

**Pull request merged**